### DISCUSSION TITLE
Streaming / Promise query("SELECT * FROM table").stream()

### DISCUSSION THREAD

**Discussion by aheissenberger** [sidorares/node-mysql2#1961] [Q&A]
It looks like that this is not possible with the promise version of this lib - what is the right way to use streaming?

I tried both but failed - as the stream() method does not exist :-(
```ts
import { createPool } from 'mysql2/promise';

  const mySQLConnection = await createConnection({
        host: 'localhost',
        database: 'eventstore',
    })
try {
        for await (const row of (await mySQLConnection.query("SELECT * FROM test_events").stream())) {
        // for await (const row of ((await mySQLConnection.query("SELECT * FROM test_events")).stream())) {
            // handle row here
        }
    } catch (e) {
        // connection errors are not caught here
        console.log('caught error', e);
    }
```

**Comment by sidorares** [comment]
currently no way unfortunately. Also not sure if easy / possible to chain `.stream()` to query. Currently query returns a promise that resolves with a full ( buffered ) result, `.stream()` would need to prevent that somehow

One possible option is to introduce another top level method, `streamQuery()` or something similar

**Comment by sidorares** [comment]
similar discussion here: https://github.com/sidorares/node-mysql2/issues/677
